bundles/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/PersistenceExtensions.java

```diff
@@ -1578,6 +1581,143 @@ private static void internalPersist(Item item, TimeSeries timeSeries, @Nullable
+     *
+     * @param item the {@link Item} to get the median value for
+     * @param timestamp the point in time to which to search for the median value
+     * @param serviceId the name of the {@link PersistenceService} to use
+     * @return the median value until <code>timestamp</code>, or <code>null</code> if no
+     *         future states could be found or if the persistence service given by <code>serviceId</code> does not
+     *         refer to an available {@link QueryablePersistenceService}. The current state is included in the
+     *         calculation.
+     */
+    public static @Nullable State medianUntil(Item item, ZonedDateTime timestamp, @Nullable String serviceId) {
+        return internalMedianBetween(item, null, timestamp, serviceId);
+    }
+
+    /**
+     * Gets the median value of the state of a given {@link Item} between two certain points in time.
+     * The {@link PersistenceService} identified by the <code>serviceId</code> is used.
+     *
+     * @param item the {@link Item} to get the median value for
+     * @param begin the point in time from which to start the summation
+     * @param end the point in time to which to start the summation
+     * @param serviceId the name of the {@link PersistenceService} to use
+     * @return the median value between <code>begin</code> and <code>end</code>, or <code>null</code> if no
+     *         states could be found or if the persistence service given by <code>serviceId</code> does not
+     *         refer to an available {@link QueryablePersistenceService}
+     */
+    public static @Nullable State medianBetween(Item item, ZonedDateTime begin, ZonedDateTime end,
+            @Nullable String serviceId) {
+        return internalMedianBetween(item, begin, end, serviceId);
+    }
+
+    private static @Nullable State internalMedianBetween(Item item, @Nullable ZonedDateTime begin,
+            @Nullable ZonedDateTime end, @Nullable String serviceId) {
+        String effectiveServiceId = serviceId == null ? getDefaultServiceId() : serviceId;
+        if (effectiveServiceId == null) {
+            return null;
+        }
+        ZonedDateTime now = ZonedDateTime.now();
+        ZonedDateTime beginTime = begin == null ? now : begin;
+        ZonedDateTime endTime = end == null ? now : end;
+
+        if (beginTime.isEqual(endTime)) {
+            HistoricItem historicItem = internalPersistedState(item, beginTime, effectiveServiceId);
+            return historicItem != null ? historicItem.getState() : null;
+        }
+
+        Iterable<HistoricItem> result = getAllStatesBetween(item, beginTime, endTime, effectiveServiceId);
+        if (result == null) {
+            return null;
+        }
+
+        Item baseItem = item;
+        if (baseItem instanceof GroupItem groupItem) {
+            baseItem = groupItem.getBaseItem();
+        }
+        Unit<?> unit = baseItem instanceof NumberItem numberItem ? numberItem.getUnit() : null;
```

Comment:
Then it will not work for DimmerItem, RollershutterItem ... anymore. In theory, also SwitchItem would work, but probably not very meaningful.
None of the other persistence extension methods have this check, so I don't see why I would introduce it here.